File: design-patterns/java/DesignPatterns/test/ar/edu/unlam/cdp/patterns/strategy/StrategyTest.java
package ar.edu.unlam.cdp.patterns.strategy;

import org.junit.Test;

public class StrategyTest {

	@Test
	public void testStrategy(){

		Report report1 = new Report(new HTMLFormatter());
		report1.outputReport();
		
		Report report2 = new Report(new PlainTextFormatter());
		report2.outputReport();
		
	}

}

File: design-patterns/cs/Patrones/Patrones.Test/StrategyTest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using NUnit.Framework;
using Patrones.Strategy;

namespace Patrones.Test
{
    [TestFixture]
    public class StrategyTest
    {
        [Test]
        public void TestBasico()
        {
            var out1 = new Report(new HTMLFormatter()).OutputReport();
            var out2 = new Report(new PlainTextFormatter()).OutputReport();

            Assert.AreNotEqual(out1, out2);
        }
    }
}
